### PR TITLE
Housing Card: Affordable Rental Units are Dwindling

### LINKED ISSUE #56: Affordable Rental Units are Dwindling
# Story Card Request

**Project: Housing**
**Card Title: Change in rental affordability**

## API Details

> Include all API calls that will need to be made, including all params and the possible
> values for the params. The more specific the better! This will help prevent back and forth
> and ultimately make for a quicker turnaround.

> If you do not have an API yet, it can be added later, but it is mandatory before a frontend
> developer works on this ticket.

## Client-side Calculations

> If the data from the API is not pre-calculated, please include all formulas here.
> e.g., `affordable = rent < monthlyIncome * 0.3`

## Data Sources

> All cards should trace back to their data origins. Include public links to those origins
> So card developers can embed the links in the card.

## Wireframes

> Include a detailed description and/or image that describes your story or visualization using
> the data described above. This can be as detailed as a wireframe, or as simple as a description
> such as a bar chart that shows variable A on the X-axis and variable B on the Y-axis.

> Once you have a wireframe, Include an embedded image of the card design by pasting an image from
> your clipboard or using the image uploader tool under this text box. This is mandatory before a
> front-end developer works on this ticket.

> Also include a link to a Figma design if you have one.

### PR DESCRIPTION
**Pull request by DingoEatingFuzz**
Closes #56 

no tests 😞It's better to prioritize finishing cards at this point.

### CODE REVIEW

**Comment by jaronheard** [comment]
@DingoEatingFuzz this is failing some tests in the component library. The tests could be updated, but it made me curious whether the changes to the label changed what the tooltip is displaying ? Do you know offhand?

**Comment by DingoEatingFuzz** [comment]
@jaronheard I fixed all the things. My changes definitely busted the tooltip. I somehow didn't notice that I overrode the tooltip by using the VictoryBar label component for the labels. 

I brought back the tooltip and addressed the tests to reflect my changes. It's all sparkly again ✨

File: packages/2018-housing-affordability/src/components/AffordableRentalUnitsDwindling/index.js [hunk 1]
```diff
@@ -1,19 +1,134 @@
...
+  width: 100%;
+`;
+
+const emphasis = css`
+  color: #000;
+`;
+
+const capitalize = str => str.length && str.split(' ')
+  .reduce((full, word) => `${full} ${word[0].toUpperCase() + word.substring(1)}`, '')
+  .trim();
```
Comment (jaronheard) [inline packages/2018-housing-affordability/src/components/AffordableRentalUnitsDwindling/index.js]: you could also use `titleCase`, which I added to formatters in the component lib!

File: packages/2018-housing-affordability/src/components/AffordableRentalUnitsDwindling/index.js [hunk 2]
```diff
@@ -1,19 +1,134 @@
...
+            options={cityOptions}
+          />}
+        </div>
+        <section>
+          {isLoading && <Loader />}
+          {selectedCityData && (<div>
+            <p>
+              {capitalize(selectedCity)} ranks <strong className={emphasis}>{selectedCityLowRank.rank}/{selectedCityLowRank.total} </strong>
+              for new units that cost <strong className={emphasis}>&lt;$800/mo</strong>
+            </p>
```
Comment (jaronheard) [inline packages/2018-housing-affordability/src/components/AffordableRentalUnitsDwindling/index.js]: @clemf gradient component goes here :)

File: packages/component-library/src/HorizontalBarChart/HorizontalBarChart.js [hunk 3]
```diff
@@ -50,19 +50,27 @@ const HorizontalBarChart = ({
   const dataHeight = (bars * barHeight) + (spaces * spaceHeight);
   const additionalHeight = padding.bottom + padding.top;
 
+  const minValue = Math.min(0, ...data.map(d => d[dataValue]));
+
+
+  const NegativeAwareTickLabel = props => (
+    <VictoryLabel dx={props.scale.y(minValue) - 20} {...props} textAnchor="end" />
+  );
```
Comment (jaronheard) [inline packages/component-library/src/HorizontalBarChart/HorizontalBarChart.js]: Nice improvements for this component! 📊

**Pull request merged**